### DISCUSSION TITLE
BRouter “Mountainous”

### DISCUSSION THREAD

**Discussion by Leeb64** [devemux86/cruiser#595] [Q&A] [answered]
In the BRouter Web Client there is the option for cyclists: Avoid gradients.
When riding a motorcycle you actually need exactly the opposite!
Gradients: ->Mountains->Curves
.......just an idea
If it goes one way, it should go in the other way....?

**Comment by devemux86** [comment]
Do you mean the downhill / uphill options in the trekking profile?

They do not appear elsewhere in the profile, so they are probably used inside the code.

Have you understood how they really work?
Have you tried them and see any difference in driving profile?

**Reply by Leeb64** [reply]
I was just playing around a bit with:
[Car-Eco.brf.txt](https://github.com/user-attachments/files/16195460/Car-Eco.brf.txt)
When changing the value in line 63(assign Uphillcost) the route changes.
the larger the value, the fewer gradients.Less than zero does nothing (negative values).
I just thought - if there is a function for avoiding inclines, you should be able to use the whole thing to prefer inclines...somehow.
This would be very helpful for planning winding routes....I think.

**Reply by devemux86** [reply]
This is a profile by @poutnikl from here?
https://github.com/poutnikl/Brouter-profiles/tree/master/CarProfiles

(not a BRouter car profile with kinematic model)

**Reply by Leeb64** [reply]
Yes.

**Reply by poutnikl** [reply, marked as answer]
You can use built-in variable downhillcostfactor and uphillcostfactor. They are by default equal to costfactor, but you can set them to any value by suitable expression.  They replace costfactor if up/downhillcutoff is crossed.

I used this approach in one on-demand profile for some crazy MTB biker  ([Trekking-hilly-path](https://raw.githubusercontent.com/poutnikl/Brouter-profiles/master/BikeProfiles/Trekking-hilly-paths.brf)), which explicitly looked for uphills and downhills.  

I use downhillcostfactor and uphillcostfactor in bike profiles generally, but just for tweeking costfactor for inclines on some surfaces. E.g. gravel or grass are not so much fun for bike climbing or downhill, if you are not hard core.

See also my [Glossary](https://github.com/poutnikl/Brouter-profiles/wiki/Glossary#d) page on GitHub.
